### ISSUE TITLE
address misaligned exception seen on ld for rs1 == rd

### ISSUE THREAD

**Issue by kumaransvyoma** [vyoma-systems/zilsd#2]
## ld misaligned with spike for aligned address: 
[ld_test.zip](https://github.com/vyoma-systems/zilsd/files/14914206/ld_test.zip)

- The test does an ld on aligned address ``0x80002000``. The test when run with different register operands does not show any misaligned address  exception but using the same address with rs1 == rd generates an exception

```
**test:**
  la s2,rvtest_data
diff_reg:
  ld s4,0(s2)
same_reg:
  ld s2,0(s2)

**disassembly:**
80000124:	ee090913          	add	s2,s2,-288 # 80002000 <rvtest_data_begin>
80000128 <diff_reg>:
80000128:	00093a03          	.insn	4, 0x00093a03
8000012c <same_reg>:
8000012c:	00093903          	.insn	4, 0x00093903
```

- For example, in the  ``ld s2, 0(s2)`` case as seen above with s2 = 0x80002000 (aligned), data from mem location s2 updates the s2 without any exception. Now for the updation of s3 register, instead of taking the old s2 (aligned); the spike takes the new s2 (from test memory) and generates an exception. 
- When the test is modified to have the test memory to also have an aligned address for the new s2 value, the exception is not seen. if the 2 loads to the s2, s3 has to happen atomically with the same address and address+4, then the spike has to be fixed 
- Sail run passes without any issue for this case

## To reproduce the issue

```bash
# compile
clang -march=rv32izilsd          -static -mcmodel=medany -fvisibility=hidden -nostdlib -nostartfiles  -T ./env/link.ld      -I ./env -mabi=ilp32  ./ld-01.S -o my.elf -DTEST_CASE_1=True -DXLEN=32
# To execute 
spike --log-commits --log spike.log  --isa=rv32iZilsd  my.elf
```

**Comment by christian-herber-nxp** [comment]
should be fixed with [9c7b983](https://github.com/NXP/riscv-isa-sim/commit/9c7b98388d04f91bf3b9d3a10982367ada34ebf5)

**Comment by lavanyajagan** [comment]
fix verified
